src/pages/ReportAvatar.tsx

```diff
@@ -1,44 +1,108 @@
-    const policyName = ReportUtils.getPolicyName(report, false, policy);
-    const avatarURL = ReportUtils.getWorkspaceAvatar(report);
+    const policyName = ReportUtils.getPolicyName(report, true, policy);
+    const isPolicyRelatedReport = !!policyName;
+    const [shouldShowDefaultGroupAvatar, setShouldShowDefaultGroupAvatar] = useState(false);
+
+    const attachment = useMemo(() => {
+        if (isNewGroupDraftAvatar) {
+            return {
+                source: shouldShowDefaultGroupAvatar ? ReportUtils.getDefaultGroupAvatar(newGroupDraft?.optimisticReportID ?? '') : newGroupDraft?.avatarUri ?? '',
+                headerTitle: (newGroupDraft?.reportName ? newGroupDraft?.reportName : ReportUtils.getGroupChatName(newGroupDraft?.participants)) ?? '',
+                originalFileName: newGroupDraft?.avatarFileName ?? '',
+            };
+        }
+
+        if (isPolicyRelatedReport) {
+            return {
+                source: UserUtils.getFullSizeAvatar(ReportUtils.getWorkspaceAvatar(report), 0),
+                headerTitle: policyName,
+                originalFileName: policy?.originalFileName ?? policy?.id ?? report?.policyID ?? '',
+            };
+        }
+
+        if (ReportUtils.isGroupChat(report)) {
+            const rootGroupReport = ReportUtils.getRootParentReport(report);
+            const source =
+                UserUtils.getFullSizeAvatar(report?.avatarUrl, 0) ??
+                UserUtils.getFullSizeAvatar(rootGroupReport?.avatarUrl, 0) ??
+                ReportUtils.getDefaultGroupAvatar(ReportUtils.getRootParentReport(report)?.reportID ?? '') ??
+                '';
+            return {
+                source,
+                headerTitle: ReportUtils.getReportName(report),
+                originalFileName: report?.avatarFileName ?? rootGroupReport?.avatarFileName ?? '',
+            };
+        }
+
+        return {
+            source: UserUtils.getFullSizeAvatar(report?.avatarUrl ?? '', 0),
+            headerTitle: ReportUtils.getReportName(report),
+            originalFileName: report?.avatarFileName ?? '',
+        };
+        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
+    }, [shouldShowDefaultGroupAvatar, newGroupDraft, report, policy]);
+
+    useEffect(() => {
+        const onError = () => {
+            setShouldShowDefaultGroupAvatar(true);
+        };
+        if (isNewGroupDraftAvatar) {
+            FileUtils.readFileAsync(newGroupDraft?.avatarUri ?? '', newGroupDraft?.avatarFileName ?? '', () => {}, onError);
+        }
+        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
+    }, []);
+
+    const onModalClose = useCallback(() => {
+        if (isNewGroupDraftAvatar) {
+            Navigation.goBack();
+            Navigation.isNavigationReady().then(() => {
+                Navigation.navigate(ROUTES.NEW_CHAT_CONFIRM);
+            });
+        } else {
```

Comment:
Can we just dismiss the modal?

Reply:
Ah I see. I think we should keep the user on the same page. But why we are doing this navigation that way? Why not `Navigation.goBack(ROUTES.NEW_CHAT_CONFIRM)`?

Reply:
Do you know why this happens?

Reply:
Can you please investigate the root cause of this one. And will passing `shouldEnforceFallback=true` fix the issue?